src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -313,6 +321,16 @@ private static void GetGenericInfo(PEModuleSymbol moduleSymbol, TypeDefinitionHa
                 mangleName = !ReferenceEquals(_name, metadataName);
             }
 
+            // when a file-local type from source is loaded from metadata, we do a best-effort check to identify it as a file type
+            // this is needed to allow EE to bind to file types from metadata, for example.
+            if (container.IsNamespace && GeneratedNameParser.TryParseFileTypeName(_name, out var displayFileName, out var ordinal, out var originalTypeName))
+            {
+                _name = originalTypeName;
+                var uncommon = GetUncommonProperties(force: true);
```

Comment:
>GetUncommonProperties(force: true); [](http://example.com/codeflow?start=31&length=35)

Can we set `_lazyUncommonProperties` directly, and remove `bool force` parameter from `GetUncommonProperties()`?

(If we need to, could we move this "file-local" code early enough in the constructor to ensure `GetCommonProperties(false)` has not already been called?)